pkg/controller/application/application_controller.go

```diff
@@ -475,10 +453,10 @@ func (c *Controller) cleanUpReleasesForApplication(app *shipperv1.Application) {
 	// maintain the invariant that we always delete oldest first (rather than
 	// failing to delete A and successfully deleting B and C in an 'A B C'
 	// history).
-	overhead := len(installedReleases) - int(*app.Spec.RevisionHistoryLimit)
+	overhead := len(installedReleases) - int(*app.Spec.RevisionHistoryLimit) + 1
```

Comment:
I think this breaks rollouts when revisionHistoryLimit is 1, since the incumbent gets nuked.

This led to an Application status as following (unrelated conditions removed):
```
status:
  conditions:
  - lastTransitionTime: 2018-09-11T21:32:17Z
    message: Transitioning from "web-serving-excellence-29e5b89b-0" to "web-serving-excellence-4508dfb0-0"
    status: "True"
    type: RollingOut
  history:
  - web-serving-excellence-4508dfb0-0
```

I didn't drop a line off the bottom, the contender was the only member of history (the incumbent had been deleted).

This is my bad for not leaving a note about the slightly suspicious "+1".

Reply:
Removing the incumbent is something that shouldn't happen indeed.

Shouldn't we then limit `RevisionHistoryLimit` to at least `2` then? I may be nitpicking here, but I'd be confused if I read `RevisionHistoryLimit` is `1` and see `2` releases in the history.

Reply:
Reverted this change but adjusted the original test which was erroneously expecting more releases to be deleted than it should. The semantics of `RevisionHistoryLimit` are unchanged; in `.status.history` will be always present the contender plus `.spec.revisionHistoryLimit` past releases. 

The latest patch also includes a safe-guard to have at least one past release present, in the case the user feels creative and specifies values lesser than `1` for `RevisionHistoryLimit`.